src/main/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFsEngine.java

```diff
@@ -492,6 +490,19 @@ static boolean excludeFromAnnotations(Genotype oldGT) {
                 && oldGT.hasGQ() && oldGT.getGQ() == 0;
     }
 
+    private List<Genotype> assignNoCalls(final GenotypesContext genotypes) {
+        final List<Genotype> returnList = new ArrayList<>();
+        for (final Genotype oldGT : genotypes) {
+            final GenotypeBuilder builder = new GenotypeBuilder(oldGT);
+            //convert GQ0s that were reblocked back to no-calls for better AN and InbreedingCoeff annotations
+            if (excludeFromAnnotations(oldGT)) {
```

Comment:
I would avoid the construction of a builder and make a new genotype by enclosing all that inside this if statement, the else simply adds the oldGT to the output list without changes.

Reply:
Since ```GenotypeContext``` implements ```List<Genotype>``` this could be a stream operation:

```
  return genotypes.stream()
                .map(g -> excludeFromAnnotations(oldGT) 
                                 ? new GenotypeBuilder(g).alleles(...).make()
                                 : g)
                .collect(Collectors.toList())
```